pkg/mysql/mysql_test.go

```diff
@@ -0,0 +1,140 @@
+package mysql
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	apiv1alpha1 "github.com/percona/percona-server-mysql-operator/api/v1alpha1"
+)
+
+func TestStatefulset(t *testing.T) {
+	const ns = "orc-ns"
+
+	cr := readDefaultCluster(t, "cluster", ns)
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "some-secret",
+			Namespace: ns,
+		},
+		StringData: map[string]string{},
+	}
+
+	tests := []struct {
+		name string
+
+		cluster    *apiv1alpha1.PerconaServerMySQL
+		initImage  string
+		configHash string
+		tlsHash    string
+		secret     *corev1.Secret
+
+		compareFile string
```

Comment:
Cross posting this comment: https://github.com/percona/percona-server-mysql-operator/pull/948/files#r2144744809

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/921/commits/abae4e2bc92a76fbf5122bc9f0a5a306a5f3e6a9